src/wallet/subcommands/wallet_cmd.rs

```diff
@@ -101,48 +108,112 @@ pub enum WalletCommands {
+                };
+                println!("{addr}");
                 Ok(())
             }
             Self::Balance { address } => {
-                let response = api.wallet_balance(address.to_string()).await?;
-                println!("{response}");
+                let balance = api.wallet_balance(address.to_string()).await?;
+                println!("{balance}");
                 Ok(())
             }
             Self::Default => {
-                let response = api
-                    .wallet_default_address()
-                    .await?
-                    .unwrap_or_else(|| "No default wallet address set".to_string());
-                println!("{response}");
+                let default_addr = if let Some(keystore) = local_keystore {
+                    crate::key_management::get_default(&keystore)?.map(|s| s.to_string())
+                } else {
+                    api.wallet_default_address().await?
+                }
+                .context("No default wallet address set")?;
+                println!("{default_addr}");
                 Ok(())
             }
-            Self::Export { address } => {
-                let response = api.wallet_export(address.to_string()).await?;
+            Self::Export {
+                address: address_string,
+            } => {
+                let StrictAddress(address) = StrictAddress::from_str(&address_string)
+                    .with_context(|| format!("Invalid address: {address_string}"))?;
+
+                let key_info = if let Some(keystore) = local_keystore {
+                    crate::key_management::export_key_info(&address, &keystore)?
+                } else {
+                    api.wallet_export(address.to_string()).await?
+                };
 
-                let encoded_key = serde_json::to_string(&LotusJson(response))?;
+                let encoded_key = serde_json::to_string(&LotusJson(key_info))?;
                 println!("{}", hex::encode(encoded_key));
                 Ok(())
             }
             Self::Has { key } => {
-                let response = api.wallet_has(key.to_string()).await?;
+                let StrictAddress(address) = StrictAddress::from_str(&key)
+                    .with_context(|| format!("Invalid address: {key}"))?;
+
+                let response = if let Some(keystore) = local_keystore {
+                    crate::key_management::find_key(&address, &keystore).is_ok()
+                } else {
+                    api.wallet_has(address.to_string()).await?
+                };
                 println!("{response}");
                 Ok(())
             }
             Self::Delete { address } => {
-                api.wallet_delete(address.to_string()).await?;
+                let StrictAddress(address) = StrictAddress::from_str(&address)
+                    .with_context(|| format!("Invalid address: {address}"))?;
+
+                if let Some(mut keystore) = local_keystore {
```

Comment:
This `if-else` proliferation makes it difficult to reason about the code. Can we have it abstracted away?

Reply:
could probably be converted into `map_or` for all the occurrences or any other more eloquent solution. 

Another option is to get rid of `else` and use `return` in the `if` block.

Reply:
Cleaned it up a bit but it's still kinda ugly. Suggestions welcome.